template.moonxml

```diff
@@ -55,8 +55,22 @@ dump = (k=ROOT, v) ->
               b "(Recursion)"
             else
               stack\push v
+              tpairs = {}
               for kk, vv in pairs v
-                dump kk, vv
+                table.insert tpairs, { kk, vv }
+              table.sort tpairs, (a, b) ->
+                a,b = a[1], b[1]
+                if type a < type b then
+                  return true
+                elseif type a > type b then
+                  return false
+                else
+                  if type a == "number"
+                    return a < b
+                  else
+                    return tostring a < tostring b
```

Comment:
Unfortunately, MoonScript associativity SUCKS hard. Have you looked at the generated code?
```lua
              if type(a < type(b)) then
                return true
              elseif type(a > type(b)) then
                return false
              else
                if type(a == "number") then
                  return a < b
                else
                  return tostring(a < tostring(b))
                end
              end
```
Parentheses are unavoidable in some cases unfortunately.
```suggestion
                if (type a) < (type b) then
                  return true
                elseif (type a) > (type b) then
                  return false
                else
                  if (type a) == "number"
                    return a < b
                  else
                    return (tostring a) < (tostring b)
```